test/Tester/CancellationTests/GrainCancellationTokenTests.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.Async;
+using Orleans.TestingHost;
+using Tester;
+using UnitTests.GrainInterfaces;
+using UnitTests.Grains;
+using Xunit;
+using UnitTests.Tester;
+
+namespace UnitTests.CancellationTests
+{
+    public class GrainCancellationTokenTests : OrleansTestingBase, IClassFixture<GrainCancellationTokenTests.Fixture>
+    {
+        private class Fixture : BaseTestClusterFixture
+        {
+            protected override TestCluster CreateTestCluster()
+            {
+                return new TestCluster(new TestClusterOptions(2));
+            }
+        }
+
+        [Theory, TestCategory("Functional"), TestCategory("BVT")]
+        [InlineData(0)]
+        [InlineData(10)]
+        [InlineData(300)]
+        public async Task GrainTaskCancellation(int delay)
+        {
+            var grain = GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var tcs = new GrainCancellationTokenSource();
+            var wait = grain.LongWait(tcs.Token, TimeSpan.FromSeconds(10));
```

Comment:
we usually by convention call anything of type Task as xxxTask. here it can be just task. This is especially important with var. People should be forced to guess what is the type of wait.